web-app/HeimScripts/heatmap/preprocess.R

```diff
@@ -0,0 +1,47 @@
+library("WGCNA")
+
+
+main <- function(aggregate=FALSE){
+  msgs = c("")
+  df <- loaded_variables[[length(loaded_variables)]]
+  good.input <- ncol(df) > 3
+  if(aggregate && good.input){
+    df <- dropEmptyGene(df)
+    aggr  <- aggregate.probes(df)
+    assign("preprocessed", aggr, envir = .GlobalEnv)
+    Discarded.rows <- nrow(df) - nrow(aggr)
+    msgs <- c( paste("Total discarded rows:",Discarded.rows))
+  }
+  else{
+    msgs <- c("Incorrect subset - in order to perform probe aggregation more than one samples are needed.")
+  }
+  list(finished=T,messages=msgs)
+}
+
+aggregate.probes <- function(df){
+  measurements <- df[,3:ncol(df)]
+  row.names(measurements) <- df[,1]
+  collapsed <- collapseRows(measurements, df[,2], df[,1], "MaxMean",
+                            connectivityBasedCollapsing = TRUE,
```

Comment:
what does this do? This is not specifically specified in the algorithms specifications doc, but maybe it is present in probe aggregation function in Advanced Analysis? If that is the case, maybe we should check with the client if this is desired and if so: update the algorithms doc?